src/IO/External/InterpolateFromCocal.cpp

```diff
@@ -0,0 +1,204 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "IO/External/InterpolateFromCocal.hpp"
+
+#include <array>
+#include <string>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+extern "C" {
+// Define the COCAL Fortran reader functions here for different id_types.
+void coc2cac_co(const int& npoints, const double* x, const double* y,
+                const double* z, const char* data_directory, const int& len_dir,
+                double* lapse, double* shift_x, double* shift_y,
+                double* shift_z, double* spatial_metric_xx,
+                double* spatial_metric_xy, double* spatial_metric_xz,
+                double* spatial_metric_yy, double* spatial_metric_yz,
+                double* spatial_metric_zz, double* extrinsic_curvature_xx,
+                double* extrinsic_curvature_xy, double* extrinsic_curvature_xz,
+                double* extrinsic_curvature_yy, double* extrinsic_curvature_yz,
+                double* extrinsic_curvature_zz, double* rest_mass_density,
+                double* spatial_velocity_x, double* spatial_velocity_y,
+                double* spatial_velocity_z, double* pressure,
+                double* specific_internal_energy);
+
+void coc2cac_ir(const int& npoints, const double* x, const double* y,
```

Comment:
I assume `_co` and `_ir` stand for "corotating" and "irrotating" respectively? Why would this code here care? Shouldn't COCAL provide a single reader routine that takes as input a file name and produces output? Ie Spectre should not need to know the details of what sort of method was used to produce this ID?